### PR TITLE
[CSV] Parse negative numbers surrounded by parens

### PR DESCRIPTION
**Pull request by tsmacdonald**
Epic: https://github.com/metabase/metabase/issues/29561

### CODE REVIEW

File: src/metabase/upload.clj [hunk 1]
```diff
@@ -73,6 +73,11 @@
 
 (def ^:private currency-regex "Supported currency signs" #"[$€£¥₹₪₩₿¢\s]")
 
+(defn- with-parens
+  "Returns a regex that matches the argument, with or without surrounding parentheses."
+  [number-regex]
+  (re-pattern (str "(" number-regex ")|(\\(" number-regex "\\))")))
```
Comment (qnkhuat) [inline src/metabase/upload.clj:79]: why do we need a version with escaped parens?

**Review by calherries — CHANGES_REQUESTED** [review]
~~Wondering why we match against escaped parens in `with-parens` but don't have any tests for them.~~

File: src/metabase/upload.clj (hunk 1 above)
Comment (calherries) [reply src/metabase/upload.clj:79]: Ah I see. This is regex so unescaped parens denote capturing groups, and the escaped parens match actual parens in the string. 

This should still work if we remove the capturing groups, and it's easier to read:
```suggestion
  (re-pattern (str number-regex "|\\(" number-regex "\\)")))
```

**Review by calherries — APPROVED** [review]
Looks good, just had one suggestion

**Pull request merged**